geoportailv3/static/js/infobar/projectionselectordirective.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * @fileoverview This file provides a "projectionselector" directive. This directive is
+ * used to insert an Projection Selector and Coordinate Display into the HTML page.
+ * Example:
+ *
+ * <app-projectionselector app-projectionselector-map="::mainCtrl.map" ></app-projectionselector>
+ *
+ * Note the use of the one-time binding operator (::) in the map expression.
+ * One-time binding is used because we know the map is not going to change
+ * during the lifetime of the application.
+ *
+ */
+goog.provide('app.projectionselectorDirective');
+goog.require('app');
+goog.require('app.projections');
+goog.require('ol.control.MousePosition');
+
+
+/**
+ * @return {angular.Directive} The Directive Object Definition.
+ * @ngInject
+ */
+app.projectionselectorDirective = function(appProjectionselectorTemplateUrl) {
+  return {
+    restrict: 'E',
+    scope: {
+      'map': '=appProjectionselectorMap'
+    },
+    controller: 'AppProjectionselectorController',
+    controllerAs: 'ctrl',
+    bindToController: true,
+    templateUrl: appProjectionselectorTemplateUrl
+  };
+};
+
+
+app.module.directive('appProjectionselector', app.projectionselectorDirective);
+
+
+
+/**
+ * @ngInject
+ * @export
+ * @constructor
+ */
+app.ProjectionselectorDirectiveController = function($scope, $document) {
+  this['projectionOptions'] = [
+    {'label': 'LUREF', 'value': 'EPSG:2169'},
+    {'label': 'Long/Lat WGS84', 'value': 'EPSG:4326'},
+    {'label': 'WGS84 UTM 32|31', 'value': 'EPSG:3263*'}
+  ];
+  this['projection'] = this['projectionOptions'][0];
+  this.mouseposition = new ol.control.MousePosition({
+    className: 'custom-mouse-coordinates',
+    target: $document.find('.mouse-coordinates')[0],
```

Comment:
Use the following in the partial (as indicated above):

```html
<div class="mouse-coordinates" ngeo-control="ctrl.mousePositionControl" ngeo-control-map="ctrl.map"></div>
```

and that in the controller constructor:

```js
  this['mousePositionControl'] = new ol.control.MousePosition({
    className: 'custom-mouse-coordinates',
    projection: this.projection['value'],
    coordinateFormat: …
  });
```

Note that there's no `target` anymore, and no call to `map.addControl`. That's taken care of by the ngeo-control directive.